sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedProcessorContextImpl.java

```diff
@@ -23,4 +24,13 @@ public String getLeaseToken() {
 
         return changeFeedObserverContext.getLeaseToken();
     }
+
+    @Override
+    public CosmosDiagnostics getDiagnostics() {
+        if (changeFeedObserverContext == null) {
+            throw new IllegalStateException("changeFeedObserverContext cannot be null!");
+        }
+
+        return changeFeedObserverContext.getFeedResponse().getCosmosDiagnostics();
```

Comment:
I wonder if there is a potential `NullPointerException` here.

Reply:
Added another null check